### PR TITLE
[Issue 7759] Support set Max Consumer on topic level.

### PR DESCRIPTION
**Pull request by zhanghaou**
Fix [https://github.com/apache/pulsar/issues/7759](https://github.com/apache/pulsar/issues/7759), and master issue [https://github.com/apache/pulsar/issues/2688](https://github.com/apache/pulsar/issues/2688)

### Motivation

Support set/get/remove maxConsumers on a topic level.

### Verifying this change
new unit test added.

### CODE REVIEW

**Comment by zhanghaou** [comment]
@codelipenghui @jianyun8023 @jiazhai PTAL. Thanks.

**Review by jianyun8023 — COMMENTED** [review]
Looks good to me.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicPoliciesTest.java [hunk 1]
```diff
@@ -590,4 +591,121 @@ public void testRemovePublishRate() throws Exception {
...
+        Assert.assertNull(consumer3);
+        consumer1.close();
+        consumer2.close();
+
+        Integer getMaxConsumers = admin.topics().getMaxConsumers(persistenceTopic);
+        log.info("MaxConsumers {} get on topic: {}", getMaxConsumers, persistenceTopic);
+        Assert.assertEquals(getMaxConsumers, maxConsumers);
+
+        admin.topics().deletePartitionedTopic(persistenceTopic, true);
+        admin.topics().deletePartitionedTopic(testTopic, true);
```
[the remark this answers is not in the document]
Comment (zhanghaou) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicPoliciesTest.java:649]: 'testTopic' was created in BeforeMethod.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 2]
```diff
@@ -2417,6 +2417,53 @@ protected void internalGetMaxProducers(AsyncResponse asyncResponse) {
...
+        }
+        checkTopicLevelPolicyEnable();
+        Optional<Integer> maxConsumers = getTopicPolicies(topicName)
+                .map(TopicPolicies::getMaxConsumerPerTopic);
+        if (!maxConsumers.isPresent()) {
+            asyncResponse.resume(Response.noContent().build());
+        } else {
+            asyncResponse.resume(maxConsumers.get());
+        }
+    }
```
[the remark this answers is not in the document]
Comment (zhanghaou) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2434]: How about making a new PR to resolve them all?

[the remark this answers is not in the document]
Comment (jianyun8023) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java:2434]: I think the new code can comply with this rule.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicPoliciesTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (jianyun8023) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicPoliciesTest.java:649]: Okay.

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (zhanghaou) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: Done.

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java [hunk 3]
```diff
@@ -153,6 +153,9 @@ public CmdTopics(PulsarAdmin admin) {
         jcommander.addCommand("get-inactive-topic-policies", new GetInactiveTopicPolicies());
         jcommander.addCommand("set-inactive-topic-policies", new SetInactiveTopicPolicies());
         jcommander.addCommand("remove-inactive-topic-policies", new RemoveInactiveTopicPolicies());
+        jcommander.addCommand("get-maxConsumers", new GetMaxConsumers());
+        jcommander.addCommand("set-maxConsumers", new SetMaxConsumers());
+        jcommander.addCommand("remove-maxConsumers", new RemoveMaxConsumers());
```
Comment (codelipenghui) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java:158]:
```suggestion
        jcommander.addCommand("get-max-consumers", new GetMaxConsumers());
        jcommander.addCommand("set-max-consumers", new SetMaxConsumers());
        jcommander.addCommand("remove-max-consumers", new RemoveMaxConsumers());
```

Comment (codelipenghui) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java:158]: And please also update the max producers as the above pattern.

Comment (zhanghaou) [reply pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdTopics.java]: Done.

**Review by codelipenghui — APPROVED** [review]

**Pull request merged**